slither/tools/read_storage/read_storage.py

```diff
@@ -216,6 +249,45 @@ def get_target_variables(self, **kwargs) -> None:
             if slot_info:
                 self._slot_info[f"{contract.name}.{var.name}"] = slot_info
 
+    def find_constant_slot_storage_type(
+        self, var: StateVariable
+    ) -> Tuple[Optional[str], Optional[int]]:
+        """
+        Given a constant bytes32 StateVariable, tries to determine which variable type is stored there, using the
+        heuristic that if a function reads from the slot and returns a value, it probably stores that type of value.
+        Also uses the StorageSlot library as a heuristic when a function has no return but uses the library's getters.
+        Args:
+            var (StateVariable): The constant bytes32 storage slot.
+
+        Returns:
+            type (str): The type of value stored in the slot.
+            size (int): The type's size in bits.
+        """
+        if not (var.is_constant and str(var.type) == "bytes32"):
+            return None
+        storage_type = None
+        size = None
+        funcs = [f for f in var.contract.functions if var in f.state_variables_read]
+        if len(funcs) == 0:
+            for c in self.contracts:
+                funcs.extend([f for f in c.functions if var in f.state_variables_read])
+        for func in funcs:
+            if func.return_type is not None:
+                ret = func.return_type[0]
+                size, _ = ret.storage_size
+                return str(ret), size * 8
+            for node in func.all_nodes():
```

Comment:
You may be able to use https://crytic.github.io/slither/slither/core/declarations/contract.html#Contract.get_functions_reading_from_variable to make this more general and not have to rely on matching these specific function names. If it only returns one function and that function has one return value that's less than or equal to 32 bytes, assume the return type is that.

Reply:
>If it only returns one function and that function has one return value that's less than or equal to 32 bytes, assume the return type is that.

For the most part I am doing this on lines 275-278. Though I could streamline it by using `Contract.get_functions_reading_from_variable`.

> to make this more general and not have to rely on matching these specific function names.

The reason I resorted to checking for the function names from the StorageSlot library is because there may be some slot (i.e., `_ROLLBACK_SLOT` in the standard ERC1967Upgrade.sol) which do not have any setter or getter functions. With the `_ROLLBACK_SLOT` example, the only function which reads from that variable is `_upgradeToAndCallUUPS` which contains the line `if (StorageSlot.getBooleanSlot(_ROLLBACK_SLOT).value)`.